### PR TITLE
[MPM] Change application name

### PR DESCRIPTION
**Pull request by ncrescenzio**
This PR changes the name of the application implementing the Material Point Method from `ParticleMechanicsApplication` to `MPMApplication`

@antonialarese 
@KratosMultiphysics/technical-committee

### CODE REVIEW

**Comment by loumalouomega** [comment]
I would update the Examples repository as well

**Comment by ncrescenzio** [comment]
> I would update the Examples repository as well

I have already prepared a branch in the `Example` and `GiDInterface` repositories with the changes required. I was thinking of opening the PR there only once this will be merged.

File: applications/CoSimulationApplication/tests/mpm_fem_beam/ProjectParametersMPM.json [hunk 1]
```diff
@@ -78,13 +78,13 @@
         }]
     },
     "print_output_process" : [{
-        "python_module"   : "particle_json_output_process",
-        "kratos_module"   : "KratosMultiphysics.ParticleMechanicsApplication",
+        "python_module"   : "mpm_json_output_process",
```
Comment (philbucher) [inline applications/CoSimulationApplication/tests/mpm_fem_beam/ProjectParametersMPM.json]: e.g. this is not backward compatible

File: applications/CoSimulationApplication/tests/mpm_fem_beam/cosim_mpm_fem_beam_parameters.json [hunk 2]
```diff
@@ -81,9 +81,9 @@
         ],
         "solvers" :
         {
-            "particle":
+            "mpm":
             {
-                "type" : "solver_wrappers.kratos.particle_mechanics_dirichlet_wrapper",
+                "type" : "solver_wrappers.kratos.mpm_dirichlet_wrapper",
```
Comment (philbucher) [inline applications/CoSimulationApplication/tests/mpm_fem_beam/cosim_mpm_fem_beam_parameters.json]: same

**Review by philbucher — COMMENTED** [review]
I dont have any objections, but I strongly recommend to add backward compatibility
Otherwise you are breaking every single existing case out there
And speaking from experience, others are not happy abt this ;)

**Comment by ncrescenzio** [comment]
@philbucher so should I duplicate all the python files, keeping both the one with the old name and the new one?

**Comment by philbucher** [comment]
I would add wrappers with the old name, that create an object of the new type. 
There you can issue deprecation warnings so that users can update at their own pace (maybe include a date in the message)

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/particle_mechanics_dirichlet_wrapper.py
Comment (ncrescenzio) [inline applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/particle_mechanics_dirichlet_wrapper.py:1]: @philbucher did you mean something like this?

Comment (ncrescenzio) [reply applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/particle_mechanics_dirichlet_wrapper.py:1]: @philbucher  can you please take a look so if you agree I can proceed with the other files?

Comment (philbucher) [reply applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/particle_mechanics_dirichlet_wrapper.py:1]: yes exactly, this looks good

File: applications/CoSimulationApplication/tests/mpm_dem/MaterialPointMaterials.json [hunk 3]
```diff
@@ -7,11 +7,11 @@
                 "name" : "LinearElasticIsotropicPlaneStrain2DLaw"
             },
             "Variables"        : {
-                "THICKNESS"             : 1.0,
-                "PARTICLES_PER_ELEMENT" : 3,
-                "DENSITY"               : 1000.0,
-                "YOUNG_MODULUS"         : 90000000.0,
-                "POISSON_RATIO"         : 0.0
+                "THICKNESS"                     : 1.0,
+                "MATERIAL_POINTS_PER_ELEMENT"   : 3,
```
Comment (philbucher) [inline applications/CoSimulationApplication/tests/mpm_dem/MaterialPointMaterials.json]: this is another silent behavior change, I dont see any migration from old to new input. You could do this in the analysis stage of the MPM-app. I.e. read the materials file and if necessary update the variable names

**Review by philbucher — COMMENTED** [review]
no objections, looking good from my side.

I am still a bit worried about the backward compatibility, seems like many things have changed

My suggestion is to only update some tests, and leave others with the old input. This way you are sure that the backward compatibility that you implement is working properly

File: applications/CoSimulationApplication/tests/mpm_dem/MaterialPointMaterials.json (hunk 3 above)
Comment (ncrescenzio) [reply applications/CoSimulationApplication/tests/mpm_dem/MaterialPointMaterials.json]: https://github.com/KratosMultiphysics/Kratos/blob/855f9891ec95d4953cd32536ab939a073efa2acb/applications/MPMApplication/python_scripts/mpm_solver.py#L493-L505

As you suggested, I added the compatibility with the old input `PARTICLES_PER_ELEMENT` here.

**Comment by ncrescenzio** [comment]
@philbucher as you suggested, I also added the wrappers with the old names of python processes and left the old inputs in the tests.

**Comment by philbucher** [comment]
cool, looks good from my side

I leave the approval to an active user of the App, maybe @VeronikaSinger ?

**Review by VeronikaSinger — APPROVED** [review]

**Comment by ncrescenzio** [comment]
Thanks @philbucher @VeronikaSinger!

**Pull request merged**